src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/MatchResult.cs

```diff
@@ -89,14 +89,14 @@ public int CompareTo(MatchResult other, string filterText)
             ImmutableArray.Create<Func<MatchResult, string, IComparable>>(
                 // Prefer the item that matches a longer prefix of the filter text.
                 (f, s) => f.RoslynCompletionItem.FilterText.GetCaseInsensitivePrefixLength(s),
+                // If there are "Abc" vs "abc", we should prefer the case typed by user.
+                (f, s) => f.RoslynCompletionItem.FilterText.GetCaseSensitivePrefixLength(s),
```

Comment:
Can we add a test that demonstrates this change in behavior?

Reply:
Added